recipe/magento2.php

```diff
@@ -98,12 +98,38 @@
 set('enable_zerodowntime', true);
 
 // Tasks
+
+// To work correctly with artifact deployment, it is necessary to set the MAGE_MODE correctly in `app/etc/config.php`
```

Comment:
Does it get higher priority than the env file?
Or it has nothing to do with the environment value and not will override. (i.e. magento ignores it)
(in fact this could have maybe an other name like DEPLOYER_MAGE_MODE) ?

Reply:
@peterjaap Am I correct about the usage of MAGE_MODE and the dev-Settings (i.e. the commands where they are used in) I was not quite sure although I know that they are required in order to work correctly.

Reply:
where can I find actually the docs/source that it is needed for deployer? and also, is it is common magento practice? where should it stay exactly, is it under system maybe?

Reply:
Well, I just wrote something but now I am less sure than before... I vaguely remember, that if you do not set the correct i.e. the target MAGE_MODE, than it will not work. Anyhow calling `setup:di:compile` without `MAGE_MODE` set also works (not sure about the outcome though) Looking up the code to find where DI compilation depends on MAGE_MODE I could not find anything. Maybe @peterjaap knows more...

Reply:
I have only experience with MAGE_MODE in the env.php file or in .htaccess (or nginx config), not yet heard of it in the config.php actually.

Reply:
MAGE_MODE has nothing to do with Deployer as such, but to actually be able to deploy artifacts correctly, Magento needs to be in production mode.

Reply:
And seeing that `env.php` files are usually not present in a build environment, adding it to `config.php` makes sense.

Reply:
@wilfriedwolf another option would be this; https://github.com/deployphp/deployer/pull/3326/files#r1009708393